src/go/rpk/pkg/cli/cmd/acl/delete.go

```diff
@@ -12,378 +12,126 @@ package acl
 import (
 	"context"
 	"fmt"
-	"sort"
-	"strings"
 
-	"github.com/Shopify/sarama"
-	"github.com/pkg/errors"
-	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
-	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/ui"
-	"golang.org/x/sync/errgroup"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
 )
 
-func NewDeleteACLsCommand(
-	admin func() (sarama.ClusterAdmin, error),
-) *cobra.Command {
-	var (
-		resource        string
-		resourceName    string
-		namePattern     string
-		operations      []string
-		allowPrincipals []string
-		allowHosts      []string
-		denyPrincipals  []string
-		denyHosts       []string
-	)
-	command := &cobra.Command{
-		Use:          "delete",
-		Short:        "Delete ACLs",
-		SilenceUsage: true,
-		Args: func(_ *cobra.Command, _ []string) error {
-			if len(allowPrincipals) == 0 &&
-				len(denyPrincipals) == 0 {
-				return fmt.Errorf(
-					"at least one of --%s or --%s must be set",
-					allowPrincipalFlag,
-					denyPrincipalFlag,
-				)
+func NewDeleteCommand(fs afero.Fs) *cobra.Command {
+	var a acls
+	var printAllFilters bool
+	cmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete ACLs",
+		Args:  cobra.ExactArgs(0),
+		Run: func(cmd *cobra.Command, _ []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			cl, err := kafka.NewFranzClient(fs, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer cl.Close()
+
+			deletions, _, err := a.createDeletionsAndDescribes(false)
+			out.MaybeDieErr(err)
+			if len(deletions) == 0 {
+				out.Exit("Specified flags result in no ACL filters for deleting.")
 			}
-			err := checkSupported(
-				supportedResources(),
-				resource,
-				resourceFlag,
-			)
-			if err != nil {
-				return err
+
+			req := kmsg.NewPtrDeleteACLsRequest()
+			req.Filters = deletions
+			resp, err := req.RequestWith(context.Background(), cl)
+			out.MaybeDie(err, "unable to issue acl deletion request: %v", err)
+			if len(resp.Results) != len(req.Filters) {
+				out.Die("response contained %d results of the expected %d!", len(resp.Results), len(req.Filters))
 			}
-			if namePattern != "" {
-				err = checkSupported(
-					supportedPatterns(),
-					namePattern,
-					namePatternFlag,
-				)
-				if err != nil {
-					return err
+
+			// If any filters failed, or if all filters are
+			// requested, we print the filter section.
+			var printFailedFilters bool
+			for _, r := range resp.Results {
+				if r.ErrorCode != 0 {
+					printFailedFilters = true
 				}
 			}
```

Comment:
I don't see such filters in createACL command. Is the delete and list command different from create command in terms of user provided flags?

Reply:
Creation doesn't use filters at all. For user-provided flags, there is some small difference in how empty values are handled, but this code specifically opts in to those differences with the new `--any` flag (only for creating / deleting). I plan to note this in the follow on large doc PR.